crates/sui/src/upgrade_compatibility/mod.rs

```diff
@@ -715,18 +715,25 @@ pub(crate) async fn check_compatibility(
     let policy =
         UpgradePolicy::try_from(upgrade_policy).map_err(|_| anyhow!("Invalid upgrade policy"))?;
 
-    compare_packages(existing_modules, new_package, package_path, policy)
+    compare_packages(
+        AccountAddress::from(package_id),
```

Comment:
This isn't quite right unless the package being upgraded is the original package.

What you want here is to get the original ID/runtime ID for the package and substitute that in.

You can do this with 
```
*p.to_move_package(u64::MAX /* safe as this pkg comes from the network */)?.original_package_id(),
```